### DISCUSSION TITLE
Hyperspi Raspberry pi 3B+ Leds Freeze

### DISCUSSION THREAD

**Discussion by Todderbod** [awawa-dev/HyperSPI#26] [Q&A]
When I first Turned them on they worked perfect but the colors were off. I made several adjustments in Hyperhdr and now my leds work for awhile and then they freeze staying on whatever screen color was on at the time it messed up. if I unplug the led power supply to the leds and the esp32 and plug it back in it works for a little while then the same thing happens over and over  This is my first setup and Im not sure what I need to do or what you need so someone can help me figure this out

**Comment by houseofquad1974** [comment]
Pretty sure @awawa-dev will be along shortly, he’s amazing when it comes to helping folk out. For now though, I would suggest listing your whole setup. Which model Pi, which version of HyperHDR, what type of LED strip. Basically, as much info as you can offer at this stage will hopefully help to diagnose your issue quicker.

**Comment by awawa-dev** [comment]
@Todderbod if they were working fine and now stopped after some changes, most likely an incorrect configuration is causing it. If you use multi-segment you must be 100% sure that the number of defined LEDs in HyperHDR is not more (ideally it should be equal) than twice SECOND_SEGMENT_START_INDEX in the firmware you are currently using, otherwise HyperSPI may be unstable. You can do an experiment: backup the HyperHDR configuration by exporting it from the web panel, then reduce the number of LEDs only in HyperHDR to some low number, e.g. SECOND_SEGMENT_START_INDEX (so it's using only the first segment), check if it works stably now.
